op-e2e/actions/proofs/pectra_blob_schedule_test.go

```diff
@@ -0,0 +1,124 @@
+package proofs_test
+
+import (
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/genesis"
+	actionsHelpers "github.com/ethereum-optimism/optimism/op-e2e/actions/helpers"
+	"github.com/ethereum-optimism/optimism/op-e2e/actions/proofs/helpers"
+	legacybindings "github.com/ethereum-optimism/optimism/op-e2e/bindings"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/predeploys"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+func TestPectraBlobSchedule(gt *testing.T) {
+	matrix := helpers.NewMatrix[any]()
+	defer matrix.Run(gt)
+
+	matrix.AddDefaultTestCases(
+		true, // use PectraBlobSchedule
+		helpers.NewForkMatrix(helpers.Holocene),
+		testPectraBlobSchedule,
+	).AddDefaultTestCases(
+		false, // don't use PectraBlobSchedule
+		helpers.NewForkMatrix(helpers.Holocene),
+		testPectraBlobSchedule,
+	)
+}
+
+func testPectraBlobSchedule(gt *testing.T, testCfg *helpers.TestCfg[any]) {
+	usePectraBlobSchedule := testCfg.Custom.(bool) // two flavors of this test
+	t := actionsHelpers.NewDefaultTesting(gt)
+	testSetup := func(dc *genesis.DeployConfig) {
+		dc.L1PragueTimeOffset = ptr(hexutil.Uint64(0))
+		if usePectraBlobSchedule {
+			// fix blob schedule after 2 L1 blocks
+			dc.L2GenesisPectraBlobScheduleTimeOffset = ptr(hexutil.Uint64(24))
```

Comment:
It would be interesting to have a test case where it activates on a timestamp that doesn't align with a L1 block timestamp, causing it to activate on L2 if misconfigured to use L2 timestamp instead of L1 origin timestamp.

Reply:
agree, added